stakepooldclient/stakepooldclient.go

```diff
@@ -664,3 +663,68 @@ func (s *StakepooldManager) GetStakeInfo() (*pb.GetStakeInfoResponse, error) {
 	}
 	return nil, errors.New("GetStakeInfo RPC failed on all stakepoold instances")
 }
+
+// comaprePubkeys runs the gRPC command WalletMasterPubKeys on all stakepoold
+// instances, performs fn on all key, value pairs of received maps, and returns
+// whether any are true. Returns an error if any gRPC commands fail.
+func (s *StakepooldManager) comparePubKeys(fn func(string, string) bool) (bool, error) {
+	for i, conn := range s.grpcConnections {
+		client := pb.NewStakepooldServiceClient(conn)
+		resp, err := client.WalletMasterPubKeys(context.Background(), &pb.WalletMasterPubKeysRequest{})
+		if err != nil {
+			return false, fmt.Errorf("GetMasterPubkeys gRPC failed on stakepoold instance %d: %v", i, err)
+		}
+		for _, v := range resp.MasterPubKeys {
+			if fn(v.Account, v.PubKey) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// WalletsHavePubKey sends gRPC commands to all stakepoold and returns whether
+// pubkey exists on any wallets. Returns an error if any gRPC commands fail.
+func (s *StakepooldManager) WalletsHavePubKey(pubkey string) (bool, error) {
+	hasPubkeyFn := func(_ string, v string) bool {
+		return v == pubkey
+	}
+	hasPubKey, err := s.comparePubKeys(hasPubkeyFn)
+	if err != nil {
+		return false, err
+	}
+	return hasPubKey, nil
+}
+
+// DefaultAccountPubKey sends gRPC commands to all stakepoold and returns the
+// master extended public key for the wallet's "default" account. Returns an
+// error if pubkeys for the "default" account are not the same accross all
+// stakepoold wallets or any gRPC commands fail.
+func (s *StakepooldManager) DefaultAccountPubKey() (string, error) {
+	var pubkey string
```

Comment:
Could rename to `validDefaultAccountPubKey` to signify that
- the pubkey of interest is for the default account (should be obvious but could become confusing if my previous suggestion is applied)
- the value stored here at the end of the day ought to be the same for all stakepoold instances (hence the `valid` prefix)

Could probably just use named return variables as well.

Reply:
jholdstock isn't a fan of naked returns.

Reply:
Could just rename the variable then

Reply:
Named return variables (as suggested by itswisdomagain) are fine. I actually kind of wish naming return values was enforced and not optional.

It's just naked returns which are icky.

Reply:
I was thinking of it as a package deal, but you mean naming them, and then placing their names in the return I guess. Never even thought about that.

@itswisdomagain sorry, my naked comments on your pr are in the wrong place.